addons/mrp/models/mrp_bom.py

```diff
@@ -114,6 +114,24 @@ def name_create(self, name):
     def name_get(self):
         return [(bom.id, '%s%s' % (bom.code and '%s: ' % bom.code or '', bom.product_tmpl_id.display_name)) for bom in self]
 
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
What if the user create a BoM as a normal BoM then write 'phantom' on it

Reply:
@amoyaux Indeed, but adding a check in `write` method would be the same as adding an api constraint, isn't it ?
I made something similar https://github.com/odoo/odoo/commit/286a527f2e56d6c46d61e8394b982e6dae8eb946 but it has been rejected (since the ValidationError would be raised each time a user changes something on an old orderpoint)

Reply:
@adwid It's better with the constraint. We were afraid that it would raise a warning on miss configuration. But it would only raise when it modify the constraint fields. They could correct it without issue. I pref the solution with constraint